packages/polaris-viz/src/components/LegendContainer/hooks/useOverflowLegend.ts

```diff
@@ -71,13 +71,15 @@ export function useOverflowLegend(props: UseOverflowLegendProps) {
     const containerWidth =
       width - leftMargin - horizontalMargin - activatorWidth;
 
-    legendItemDimensions.current.reduce((totalWidth, card, index) => {
-      if (totalWidth + card.width + index * LEGEND_GAP > containerWidth) {
-        lastVisibleIndex = index;
-      } else {
-        return totalWidth + card.width;
-      }
-    }, lastVisibleIndex);
+    legendItemDimensions.current
+      .filter((dimension) => Boolean(dimension))
```

Comment:
I need to check on this because for some reason there is an empty slot on the legendItemDimensions array 🤔